File: paulkim1997/주식가격.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    //주식 가격 만큼 반복
    for(int i=0;i<prices.size();i++) {
        int price = 0;
        //i번째 주식 가격에 대해 i+1번 부터 끝까지 비교해
        //가격이 언제 떨어지는지 체크
        for(int j=i+1;j<prices.size();j++) {
            //가격이 안떨어진다면 1 더해주면서 지속
            if(prices[i] <= prices[j])
                price++;
            //가격이 떨어졌다면 1초 더해주고 빠져나옴
            else {
                price++;
                break;
            }
        }
        answer.push_back(price);

    }
    return answer;
}

File: bsw112/2021-02-01/주식가격.java
public class 주식가격 {

    //https://programmers.co.kr/learn/courses/30/lessons/42584
    //초당 가격이 주어진다.
    //가격이 하락하지 않은 동안의 시간을 초로 구하라.

    // 1 ,2, 3, 2, 3 이면 맨처음 1은 5초간 유지. 2는 4초간, 3은 1초간, 2는 2초간, 3은 0초간
    public static void main(String[] args) {
        주식가격 module = new 주식가격();
        System.out.println(module.solution(new int[]{1, 2, 3, 2, 3}));
    }

    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        for (int i = 0; i < prices.length; ++i) {
            int cnt = 0;
            for (int j = i + 1; j < prices.length; ++j) {
                cnt = j - i;
               if(prices[j] < prices[i])
                   break;

            }

            answer[i] = cnt;
        }

        return answer;
    }

}
